core/src/main/java/org/htsjdk/core/utils/ValidationUtils.java

```diff
@@ -0,0 +1,77 @@
+package org.htsjdk.core.utils;
+
+import java.util.Collection;
+import java.util.function.Supplier;
+
+/**
+ * General utilities to check and validate conditions for constructors/methods.
+ */
+public final class ValidationUtils {
+
+    // cannot be instantiated
+    private ValidationUtils() {}
+
+    /**
+     * Checks that the condition is {@code true}.
+     *
+     * @param condition required condition.
+     * @param message   the text message that would be passed to the exception thrown.
+     */
+    public static void validateArg(final boolean condition, final Supplier<String> message) {
+        if (!condition) {
+            throw new IllegalArgumentException(message.get());
+        }
+    }
+
+    /**
+     * Checks whether an index is within bounds.

```

Comment:
I might reword to "Checks whether an index is within the bounds  [0, length).

Reply:
Done.